### ISSUE TITLE
incorrect sizeof long for 32-bit apple arm

### ISSUE THREAD

**Issue by ehaas** [Vexu/arocc#810]
@alexrp I *think* this Zig commit: https://github.com/ziglang/zig/commit/b94bb6f96f475efcdb24c64fe36a4ec22bf18cdf broke aro's record layout tests. I say "think" because I haven't actually bisected, but just looking at the code it seems likely that it would introduce the behavior I'm seeing. If aro is built with a recent master build of the Zig compiler:

`arocc --target=arm-ios-none test.c`
```test.c
_Static_assert(sizeof(long) == 4, ""); // Fails; passes if you change 4 to 8
```

Based on the commit message it sounds like 32-bit arm ios is dead / unsupported; should we remove it from our test matrix? Explicitly set some other abi than `none`?

**Comment by alexrp** [comment]
Yes, I suggest removing any coverage of Apple + 32-bit Arm. It's a very dead target that Zig has never properly supported to my knowledge. The last iPhone using 32-bit Arm was released in 2013, and the last Apple Watch using it was released in 2017. If you look [here](https://developer.apple.com/support/xcode), you can see that `armv7` and `armv7s` were removed in Xcode 14 (2022), with `armv7k` presumably soon to follow in the next couple of years.

**Comment by alexrp** [comment]
> Apple + 32-bit Arm

Oh, and by the way, this shouldn't be confused with Arm64 + 32-bit ABI. This is called `aarch64_32` in LLVM/Clang but represented as `aarch64-watchos-ilp32` in Zig. This is still a relevant target.

**Comment by ehaas** [comment]
Cool. Indeed the tests are for armv7, they're ultimately derived from here https://github.com/mahkoh/repr-c/blob/master/repc/impl/build.rs#L53

@Vexu are you ok with me removing that from the record layout tests? The alternative would be to handle it as a special case in the size/alignment functions.

**Comment by Vexu** [comment]
> are you ok with me removing that from the record layout tests?

Yes, no point supporting a target Apple doesn't support.

**Comment by ehaas** [comment]
Closed by #812
